benchkit/shell/shellasync.py

```diff
@@ -142,6 +181,10 @@ def stop(self) -> None:
 
         # Kill assynchronous process and all processes on the same group id
         # As we create a new group id for each background process, this will kill its children
+
+        remote_pid = self.find_matching_ssh(self._process.pid)
```

Comment:
are you sure it's always through ssh? Isn't this a generic shellasync that could locally as well?

Reply:
You're correct. I need to first check whether or not this is remote

Reply:
Added a code to check whether the call is remote or not